admin/apple-actions/index/class-push.php

```diff
@@ -540,16 +527,83 @@ private function process_errors( $errors ) {
 			// Throw an exception.
 			throw new Action_Exception( esc_html( $alert_message ) );
 		} elseif ( 'warn' === $component_alerts && ! empty( $errors[0]['component_errors'] ) ) {
-			\Admin_Apple_Notice::error( $alert_message, $user_id );
+			Admin_Apple_Notice::error( $alert_message, $user_id );
 		}
 	}
 
+	/**
+	 * Allows debugging of API requests by dumping the data
+	 * that will be sent to the API.
+	 */
+	public function debug(): void {
+		$post_id = $this->id;
+		$post    = get_post( $post_id );
+		if ( ! $post ) {
+			return;
+		}
+
+		$this->sections              = Admin_Apple_Sections::get_sections_for_post( $post_id );
+		[ $json, $bundles, $errors ] = $this->generate_article();
+
+		// Retrieve the metadata using the same method as in the push() function.
+		$meta = [ 'data' => [] ];
+
+		// Set sections.
+		if ( ! empty( $this->sections ) ) {
+			sort( $this->sections );
+			$meta['data']['links'] = [ 'sections' => $this->sections ];
+		}
+
+		// Get the various settings.
+		$is_paid         = (bool) get_post_meta( $post_id, 'apple_news_is_paid', true );
+		$is_preview      = (bool) get_post_meta( $post_id, 'apple_news_is_preview', true );
+		$is_hidden       = (bool) get_post_meta( $post_id, 'apple_news_is_hidden', true );
+		$is_sponsored    = (bool) get_post_meta( $post_id, 'apple_news_is_sponsored', true );
```

Comment:
in 2.5.0, these actually can be set by the meta or inherit the channel default, so they won't always be `bool` any more. See https://github.com/alleyinteractive/apple-news/pull/1090/files

Reply:
Great, will update.

Reply:
Give this a look, thanks Greg.